extensions/integration-tests/src/notebook.test.ts

```diff
@@ -14,136 +14,183 @@ import { sqlNotebookContent, writeNotebookToFile, sqlKernelMetadata, getFileName
 			test('Python3 notebook test', async function () {
-				let notebook = await openNotebook(pySparkNotebookContent, pythonKernelMetadata, this.test.title);
-				let cellOutputs = notebook.document.cells[0].contents.outputs;
-				console.log('Got cell outputs ---');
-				if (cellOutputs) {
-					cellOutputs.forEach(o => console.log(o));
-				}
-				let result = (<azdata.nb.IExecuteResult>cellOutputs[0]).data['text/plain'];
-				assert(result === '2', `Expected python result: 2, Actual: ${result}`);
+				await (new NotebookTester()).python3NbTest(this.test.title);
 			});
 
 			test('Clear all outputs - Python3 notebook ', async function () {
-				let notebook = await openNotebook(pySparkNotebookContent, pythonKernelMetadata, this.test.title);
-				await verifyClearAllOutputs(notebook);
+				await (new NotebookTester()).python3ClearAllOutputs(this.test.title);
 			});
 		}
 
 		if (process.env['RUN_PYSPARK_TEST'] === '1') {
 			test('PySpark3 notebook test', async function () {
-				let notebook = await openNotebook(pySparkNotebookContent, pySpark3KernelMetadata, this.test.title);
-				let cellOutputs = notebook.document.cells[0].contents.outputs;
-				let sparkResult = (<azdata.nb.IStreamResult>cellOutputs[3]).text;
-				assert(sparkResult === '2', `Expected spark result: 2, Actual: ${sparkResult}`);
+				await (new NotebookTester()).pySpark3NbTest(this.test.title);
 			});
 		}
 	});
 }
 
-async function openNotebook(content: azdata.nb.INotebookContents, kernelMetadata: any, testName: string, runAllCells?: boolean): Promise<azdata.nb.NotebookEditor> {
-	let notebookConfig = vscode.workspace.getConfiguration('notebook');
-	notebookConfig.update('pythonPath', getConfigValue(EnvironmentVariable_PYTHON_PATH), 1);
-	let server = await getBdcServer();
-	assert(server && server.serverName, 'No server could be found in openNotebook');
-	await connectToServer(server, 6000);
-	let notebookJson = Object.assign({}, content, { metadata: kernelMetadata });
-	let uri = writeNotebookToFile(notebookJson, testName);
-	console.log(uri);
-	let notebook = await azdata.nb.showNotebookDocument(uri);
-	console.log('Notebook is opened');
-
-	if (!runAllCells) {
-		assert(notebook.document.cells.length === 1, 'Notebook should have 1 cell');
-		console.log('Before run notebook cell');
-		let ran = await notebook.runCell(notebook.document.cells[0]);
-		console.log('After run notebook cell');
-		assert(ran, 'Notebook runCell should succeed');
-	} else {
-		console.log('Before run all notebook cells');
-		let ran = await notebook.runAllCells();
-		assert(ran, 'Notebook runCell should succeed');
-		assert(notebook !== undefined && notebook !== null, 'Expected notebook object is defined');
+class NotebookTester {
+	invocationCount: number = 0;
+
+	private static IterationCount = 20;
+	private static ParallelCount = 1;
+
+	@stressify({ dop: NotebookTester.ParallelCount, iterations: NotebookTester.IterationCount })
+	async pySpark3NbTest(title: string) {
```

Comment:
Please make sure all functions have a return value - or void in most of these cases.